### ISSUE TITLE
No ads get blocked in the First opened Incognito window. Second and subsequent tabs all have adblock

### ISSUE THREAD

**Issue by 3ICE** [uBlockOrigin/uBlock-issues#414]
Hi,

I like to check risky links in incognito mode first. Risky links can be anything from unknown youtube videos (so I don't get RickRoll'd in my watch history), torrents, new flash/html5 games, to potential 18+ content, even. Those kinds of websites are the most heavily spammed with advertisements, usually. So that is when I would need uBlock's protection the most, but that is also when it fails me... Sometimes. The website ads are able to load before adblocking is ready, it seems. And upon loading it does not retroactively zap all already loaded adverts and their artifacts.
Subsequent incognito tabs are adblocked fine. Only the first one fails to react in time. Only reproducible the first time in between restarts.

Obviously in Chrome's extension settings for uBlock I have "allow in incognito mode" checked. And it does load after a couple hundred milliseconds of delay in the incognito window too. But by then it is too late. Ads are rendering in...

Workaround: Leave an incognito window open so the extension is always ready. (Not feasible; I want those sessions to be as short lived as possible. Incognito cookies only disappear once all incognito windows are closed.)

I also filled out this template thing below, though I am not a fan of it. Two notes:
- Logger is not accessible immediately upon window open. By the time I manage to open it (roughly one second reaction time) all issues are already present on the page and the logger is still empty. Reloading from the logger is of course not applicable because it fixes the issue and blocks all ads as expected.
- *See ads on the worst offenders only by turning off Chrome's built in ad blocker: Settings, Advanced, Content Settings, Ads, toggle "Blocked on sites that tend to show intrusive ads" to "Allow Ads"

Sincerely,
Daniel "3ICE" Berezvai
[3ice.hu](http://3ice.hu/)

Do NOT delete this template or any part of it when submitting your issue - okay

### Prerequisites

Check the appropriate boxes after you submit your issue - done
Speculated performance issues will be marked as invalid and closed if they do not come with actual profiling data + analysis supporting the claim
Opening issues for adding new filter lists is now disallowed and such issues will be declined and closed

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
        - If disabling uBO <https://github.com/gorhill/uBlock/wiki/Quick-guide:-popup-user-interface#the-large-power-button> makes the issue go away, then in all likelihood this is a filter issue.
        - See what the logger <https://github.com/gorhill/uBlock/wiki/The-logger> reports when you reproduce the issue, this will help you determine whether this is a filter issue.
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/) Such issue will be closed as invalid
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported. - nope
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile *
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Extension loads only AFTER ads are loaded. In the first incognito window.
Gets me only once per reboot / after considerable amount of waiting.

### A specific URL where the issue occurs
https://www.youtube.com/watch?v=j6WYgxUVrc0
or
https://1337x.to/search/ubuntu/1/ *

### Steps to Reproduce

0. Have no incognito windows open for a while
1. Right click a link
2. Open in incognito
3. Observe ads pop up

### Expected behavior:

No ads should load

### Actual behavior:

Ads are not caught in time.
Extension uBlock seems to have loaded too late to prevent them.

### Your environment

* uBlock Origin version: 1.18.2
* Browser Name and version: Google Chrome Version 71.0.3578.98 (Official Build) (64-bit)
* Operating System and version: Microsoft Windows 10 Pro Version 10.0.17134 Build 17134

**Comment by gorhill** [comment]
Duplicate of <https://github.com/gorhill/uBlock/issues/1327>.

See <https://github.com/gorhill/uBlock/wiki/Advanced-settings#suspendtabsuntilready-experimental>.
